source-hubspot-native/source_hubspot_native/api.py

```diff
@@ -484,6 +484,21 @@ async def fetch_recent_email_events(
     if yielded_event:
         yield max_ts + timedelta(milliseconds=1) # startTimestamp is inclusive.
 
+async def _fix_formats(
+    log: Logger,
+    http: HTTPSession,
+    object_name: str,
+    doc: CRMObject,
+) -> CRMObject:
+    property_types = {p.name: p.type for p in (await fetch_properties(log, http, object_name)).results}
```

Comment:
this looks ... expensive. I recognize it's cached, but we're still building a coroutine and new dict entry for every property of every document

Stepping back, can we do the really basic thing of filtering out any empty property value from the documents properties, regardless of its type? This seems just as much a possible problem for properties which are string-formatted numbers, or emails, or other things.

Is it a problem for regular string properties having empty values to map to NULL ?

Reply:
The only slightly weird thing that could happen if we completely filter a property from the properties object is that it would still show up in the property history as an empty string. I don't know if that is a problem or not...probably not.

And I guess I'm not specifically aware of any problem that would arise from filtering all empty string properties regardless of their types, but I can't say with 100% confidence that is true. I was thinking of a case where there is some difference to the user between `""`, `"something"`, and the complete absence of the thing. But I do think you are right that formatted numbers would also have this problem. So, doing the really basic thing you suggest is probably fine, and we can revisit later if it's an issue.